polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -3469,25 +3489,250 @@ void MLIRASTConsumer::setMLIRFunctionVisibility(
   SymbolTable::setSymbolVisibility(function, visibility);
 }
 
+/// Determines whether the language options require us to model
+/// unwind exceptions.  We treat -fexceptions as mandating this
+/// except under the fragile ObjC ABI with only ObjC exceptions
+/// enabled.  This means, for example, that C with -fexceptions
+/// enables this.
+static bool hasUnwindExceptions(const LangOptions &LangOpts) {
+  // If exceptions are completely disabled, obviously this is false.
+  if (!LangOpts.Exceptions)
+    return false;
+
+  // If C++ exceptions are enabled, this is true.
+  if (LangOpts.CXXExceptions)
+    return true;
+
+  // If ObjC exceptions are enabled, this depends on the ABI.
+  if (LangOpts.ObjCExceptions)
+    return LangOpts.ObjCRuntime.hasUnwindExceptions();
+
+  return true;
+}
+
+void MLIRASTConsumer::setMLIRFunctionAttributesForDefinition(
+    const Decl *D, mlir::FunctionOpInterface F) const {
+  const CodeGenOptions &CodeGenOpts = CGM.getCodeGenOpts();
+  const LangOptions &LangOpts = CGM.getLangOpts();
+  MLIRContext *Ctx = module->getContext();
+  mlirclang::AttrBuilder B(*Ctx);
+
+  if (CodeGenOpts.UnwindTables)
```

Comment:
`SetLLVMFunctionAttributesForDefinition` has the following:
```suggestion
  if ((!D || !D->hasAttr<NoUwtableAttr>()) && CodeGenOpts.UnwindTables)
```

Reply:
Must be something that has been updated after we branched. Added.